src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildJobExecutionService.java

```diff
@@ -453,21 +456,45 @@ private BuildResult constructBuildResult(List<BuildResult.LocalCITestJobDTO> fai
     }
 
     private Path cloneRepository(VcsRepositoryUri repositoryUri, @Nullable String commitHash, boolean checkout, String buildJobId) {
+        int attempt = 0;
+        Repository repository = null;
+
+        // Ignore warning about always true condition, as the loop will break if the clone is successful
+        while (attempt < MAX_CLONE_RETRIES) {
+            try {
+                attempt++;
+                // Clone the assignment repository into a temporary directory
+                // TODO: use a random value if commitHash is null
```

Comment:
Comment to track this TODO

Reply:
I am not sure what you mean

Reply:
You have a TODO here and I adde a comment to track this. This is a pretty easy thing to change, so I assumed this was intended to be done in this PR.

Reply:
Done, added handling of the case when commit hash is null